src/Artsy/Analytics/v2/Events/AuthImpression.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  ActionType,
+  AuthContextModule,
+  AuthIntent,
+  AuthModalType,
+  AuthTrigger,
+} from "../Schema"
+
+interface AuthImpression {
+  action: ActionType.authImpression
+  context_module: AuthContextModule
+  intent: AuthIntent
+  modal_copy?: string
+  onboarding: boolean
+  trigger: AuthTrigger
+  trigger_seconds?: number
+  type: AuthModalType
+}
+
+interface AuthImpressionArgs {
+  copy?: string
+  contextModule: AuthContextModule
+  intent: AuthIntent
+  onboarding?: boolean
+  trigger?: AuthTrigger
+  triggerSeconds?: number
+  type: AuthModalType
+}
```

Comment:
Is there any way of avoiding some of the repetition here? The `AuthImpression` interface and the args declaration are very similar.

Reply:
We gain two things from using a different interface for args - some options can be optional and have default values, and the naming convention. In JS apps our convention uses `contextModule` for args, and this format is ideal. Segment expects `context_module`, so the keys are different on the `AuthImpression` event interface.